### DISCUSSION TITLE
The current modules are too few/limited

### DISCUSSION THREAD

**Discussion by jpgpng** [ringo/ringojs#464] [Ideas & Feedback]
Have a look at fantom's: https://fantom.org/doc/

Will we have something like this? Or resort to calling Java is the answer? Yes, it's possible to call Java libraries. Fantom is also Java based. But having ready to use modules/wrappers is much better.

**Comment by botic** [comment]
The way to go is calling Java libraries directly. There are no plans to provide more modules in the core. We reduced the number of core modules in the last releases and marked some as legacy: https://github.com/ringo/legacy-modules

**Comment by jpgpng** [comment]
> The way to go is calling Java libraries directly. There are no plans to provide more modules in the core. We reduced the number of core modules in the last releases and marked some as legacy: https://github.com/ringo/legacy-modules

I'm thinking a bit different. The less you have to call Java the better. It's like the Fantom language I mentioned above. They wrapped the Java APIs in native easy to use Fantom APIs. You will write mostly in pure Fantom code. I think the same should be applied to Ringo.

**Comment by botic** [comment]
It really depends on your use case. I'm speaking for the Ringo core itself and we want to reduce it to the "minimum viable library" for a JavaScript runtime. **Ringo itself should be non-opinionated on your approach.** In practice we see a lot of plain JS packages directly working in Ringo (with minimum effort). Heavier lifting should be done in Java / JVM, since the underlying Rhino engine is not the fastest. If you have heavy object manipulation, Rhino is definitely too slow and imho outsourcing such work to a fast Java implementation is a good idea.
